test/checks/aria/has-widget-or-window-role.js

```diff
@@ -0,0 +1,118 @@
+describe('has-widget-or-window-role', function () {
+  'use strict';
+
+  let node;
+  const fixture = document.getElementById('fixture');
+  const checkContext = axe.testUtils.MockCheckContext();
+  const evaluate = currentNode =>
+    axe.testUtils
+      .getCheckEvaluate('has-widget-or-window-role')
+      .call(checkContext, currentNode);
+  const roles = {
```

Comment:
This feels a little brittle. Generally what we do is we have the unit tests just check the code works. One or two examples of each role type is enough. And then we have fairly extensive integration tests. Those are a little more telling.

Reply:
I'll be reverting this change since the desire is to not change this check.

Though I believe this change was not brittle. This was a refactor from ~500 lines of tests to ~100 that checked the same things, but it was difficult to parse by a person before. I found myself getting quite lost in the code previously. The underlying test code was the same with this change.